versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/UpdateableObjs.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.common.persist;
+
+import static org.projectnessie.versioned.storage.common.objtypes.GenericObj.VERSION_TOKEN_ATTRIBUTE;
+
+import java.util.Optional;
+import org.projectnessie.versioned.storage.common.objtypes.GenericObj;
+
+public final class UpdateableObjs {
```

Comment:
While working on this I realized that `UpdateableObj` is not declared under `org.projectnessie.versioned.storage.common.objtypes` like all other object types. Should we move it there? I don't think there would be repercussions on stored objects because `UpdateableObj` is just a super interface of those stored objects.

Reply:
Guess that's okay.
But then the static function here could go into `UpdateableObj` as well.

Reply:
Yes - and we could turn `GenericObj` package-private again.